src/main/scala/beam/scoring/BeamScoringFunctionFactory.scala

```diff
@@ -189,10 +189,14 @@ class BeamScoringFunctionFactory @Inject() (
             .filter(_.isInstanceOf[Activity])
             .map(_.asInstanceOf[Activity])
             .lift(tripIndex + 1)
+          val tripOrigin = person.getSelectedPlan.getPlanElements.asScala
+            .filter(_.isInstanceOf[Activity])
+            .map(_.asInstanceOf[Activity])
```

Comment:
You could use `collect` instead of `filter`, `map`

Reply:
right, `collect` looks faster and cleaner to me